src/main/java/qna/domain/Question.java

```diff
@@ -36,6 +43,24 @@ public Question(Long id, String title, String contents) {
         this.contents = contents;
     }
 
+    public List<DeleteHistory> deleteQuestionAndAnswers(User loginUser) {
```

Comment:
개인적으로 퍼블릭 메서드 명을 정할때는 협력에 맞게 추상적인 작성하는 것을 선호하는데요.
그 이유는 내부 구현이 바뀌더라도, 메서드 명을 유지할 수 있기 때문입니다. 
메서드 명을 유지한다는 것은 생각보다 단순해보이기도하지만, 처음 정의한 협력을 바꾸지 않는다는 관점에서 바라보면 굉장히 중요하게 느껴지기도 하네요 🤔 

조금 극단적이게 생각해보면 질문을 삭제할때 답변을 삭제하지 않도록 요구사항이 변경될 수도 있을 것 같아요 😄 


> 구현과 관련된 모든 정보를 캡슐화하고 객체의 퍼블릭 인터페이스에는 협력과 관련된 의도만을 표현해야 한다.

```suggestion
    public List<DeleteHistory> delete(User loginUser) {
```

Reply:
네, 의도를 잘 이해 했어요. 
제 의도는 미션에 한하여 생각했고, 협력과 관계없이 메서드명 구체적이면 그것만 봐도 무엇을 할 수 있기에 그렇게 작성을 했어요.
만약 협력을 필요로 한다는 생각이 조금이라도 들었다면, 추상화 하는 것에 동의합니다.